test/multiple-user-principal-types.test.js

```diff
@@ -54,338 +69,295 @@ describe('Multiple users with custom principalType:', function() {
-          OneUser.login(commonCredentials),
-          AnotherUser.login(commonCredentials),
-        ]);
-      })
-      .then(function() {
-        return anotherUserFromOneModel.updateAttributes({email: 'updated@example.com'});
-      })
-      .then(function() {
-        return Promise.all([
-          AccessToken.find({where: {userId: anotherUserFromOneModel.id,
-            principalType: OneUser.modelName}}),
-          AccessToken.find({where: {userId: userFromOneModel.id,
-            principalType: OneUser.modelName}}),
-          AccessToken.find({where: {userId: userFromAnotherModel.id,
-            principalType: AnotherUser.modelName}}),
-        ]);
-      })
-      .spread(function(anotherUserFromOneModelTokens, userFromOneModelTokens,
-      userFromAnotherModelTokens) {
-        expect(anotherUserFromOneModelTokens.length).to.equal(0);
-        expect(userFromOneModelTokens.length).to.be.above(0);
-        expect(userFromAnotherModelTokens.length).to.be.above(0);
-        done();
-      })
-      .catch(done);
+    it('should only impact related user', function() {
+      return OneUser.create({email: 'original@example.com', password: 'bar'})
+        .then(function(u) {
+          anotherUserFromOneModel = u;
+          return Promise.all([
+            OneUser.login({email: 'original@example.com', password: 'bar'}),
+            OneUser.login(commonCredentials),
+            AnotherUser.login(commonCredentials),
+          ]);
+        })
+        .then(function() {
+          return anotherUserFromOneModel.updateAttributes({email: 'updated@example.com'});
+        })
+        .then(function() {
+          return AccessToken.find({'order': 'principalType ASC'});
+        })
+        .then(function(allTokens) {
+          var data = allTokens.map(function(token) {
+            return {userId: token.userId, principalType: token.principalType};
+          });
+          expect(data).to.eql([
+            // no token for anotherUserFromOneModel
+            {userId: userFromAnotherModel.id, principalType: 'AnotherUser'},
+            {userId: userFromOneModel.id, principalType: 'OneUser'},
+          ]);
+        });
     });
   });
 
-  describe('role model:', function() {
+  describe('AccessContext', function() {
+    var ThirdUser, userFromThirdModel, accessContext;
+
+    // helper
+    function addPrincipal(principal) {
+      if (!principal) return;
+      accessContext.addPrincipal(principal.type, principal.id);
+    }
+
+    beforeEach(function() {
+      accessContext = new AccessContext({registry: OneUser.registry});
+    });
+
+    it('getUserId() should return userId although principals contain non USER principals',
+    function() {
+      return Promise.try(function() {
+        var principals = [{type: Principal.ROLE}, {type: Principal.APP}, {type: Principal.SCOPE},
+          {type: OneUser.modelName, id: userFromOneModel.id}];
+        principals.map(addPrincipal);
```

Comment:
i personally find the two equivalents, let's pick yours to accelerate the review